pkg/githubclient/githubclient.go

```diff
@@ -33,20 +32,13 @@ type GitHub struct {
 
 // New creates a new instance of a GitHub client.
 func New(ctx context.Context, appID, rsaPrivateKeyPEM string) (*GitHub, error) {
-	// Read the private key.
-	privateKey, err := readPrivateKey(rsaPrivateKeyPEM)
+	app, err := githubauth.NewApp(appID, "", rsaPrivateKeyPEM, githubauth.WithJWTTokenCaching(1*time.Minute))
```

Comment:
Can we generalize this by accepting the `httpClient *http.Client` to pass to `github.NewClient`? Either via func signature or options pattern.

Reply:
I'd prefer to refactor that separately since:

1. I don't know much about how GMA is structured

2. It's not really related to this change (although it touches the same files)

Reply:
SGTM